### PR TITLE
feat: banner component

### PR DESCRIPTION
**Pull request by ipeglin**
⚠️ Merge only after #73 

## Changes
* New home/Banner component to greet users at landing page
* Conditional rendering of image depending on the use of mobile device for browsing. _Might refactor or change later_.

## Remaining work or improving features
* Add call-to-action button to navigate to program. This should be passed down in props.
* Add anchor to "companies" further down on the landing page

## How to use
Takes all props within a `content` object of type:
```
content: {
  date: {
    start: string //YYYY-MM-DD
    end: string //YYYY-MM-DD
  }
  title: string
  caption: string
  img: string
}
```

### Dekstop (vw > 768px)
image

### Mobile (vw <= 768px)
image

### CODE REVIEW

File: components/home/Banner.vue [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    min-width: 310px;
+    padding-block: 1rem;
+    width: 100%;
+
+    // All direct descendants
+    > * {
+      max-width: 25%;
+    }
+
+    @media (max-width: 768px) {
```
Comment (AasmundN) [inline components/home/Banner.vue:67]: I think we should use the Vuetify breakpoints to make sure everything is consistent. 

```scss
@use 'vuetify/settings';

@media #{map-get(settings.$display-breakpoints, 'sm-and-down')} {
   // styles for sm breakpoint and down 
}
```

File: pages/index.vue [hunk 2]
```diff
@@ -2,6 +2,15 @@
   <div>
     <p>User from store: {{ user }}</p>
```
Comment (AasmundN) [inline pages/index.vue:3]: Think we can remove this, as well as the create user button and script tag.

File: components/home/Banner.vue [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          !isSameMonth(content.date.start, content.date.end)
+            ? $t(`datetime.months.${startDate.month}`)
+            : ''
+        }}
+        - {{ endDate.day }}. {{ $t(`datetime.months.${endDate.month}`) }}
+      </h3>
+      <h1>{{ content.title }}</h1>
+      <h4>{{ content.caption }}</h4>
+    </div>
+    <v-img
```
Comment (AasmundN) [inline components/home/Banner.vue:43]: PascalCase

File: components/home/Banner.vue [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  )
+  const endDate: DateStringObject = dateStringToStringObject(
+    content.value.date.end
+  )
+</script>
+
+<template>
+  <div class="home-banner">
+    <div>
+      <h3>
```
Comment (AasmundN) [inline components/home/Banner.vue:31]: Might want to use Vuetify text size classes.

File: components/home/Banner.vue [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      image: string
+      title: string
+    }
+  }>()
+
+  const { mobile } = useDisplay()
+
+  // Convert event dates into readable and usable format
+  const { content } = toRefs(props)
+  const startDate: DateStringObject = dateStringToStringObject(
```
Comment (AasmundN) [inline components/home/Banner.vue:20]: Could use a computed property here to preserve reactivity, but I guess it's not really necessary.

File: components/home/Banner.vue [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    align-items: center;
+    display: flex;
+    gap: 4rem;
+    justify-content: center;
+    min-width: 310px;
+    padding-block: 1rem;
+    width: 100%;
+
+    // All direct descendants
+    > * {
```
Comment (AasmundN) [inline components/home/Banner.vue:63]: Cool

**Review by AasmundN — CHANGES_REQUESTED** [review]
Looking good. Maybe the text should be a tad bit larger? Also could we find a way to include the image on mobile view too? Might be hard to find a way that looks good I guess.

**Comment by ipeglin** [comment]
> Looking good. Maybe the text should be a tad bit larger? Also could we find a way to include the image on mobile view too? Might be hard to find a way that looks good I guess.
> 
> One thing to keep in mind when using the `useDisplay` composable is that the server has no way of knowing much about the users device. So server side rendering depending on the composable will not really work. For some things this is fine, as long as one keeps it mind. Maybe by adding some transitions. In this case I think it would be better to use a @media query.

Made the required changes with the exception of the `useMobile` composable. I haven't really looked into the use of vuetify yet, so it would be nice if you could walk me through it in person some time.

**Review by AasmundN — APPROVED** [review]
I have made a few minor changes: 
- Added a border shadow to the image, which makes it pop out a bit more, and rounded its corners. 
- Changed the header elements so that the text will be smaller. Also added the Vuetify text size classes to these elements in `main.scss`.
- Fixed the locale switch page transition problem from #99 

Think this is good to go now.

**Comment by ipeglin** [comment]
I'll add the call-to-action and anchor nav features to new issues, and merge this PR

**Pull request merged**